src/olympia/blocklist/migrations/0031_auto_20230612_1447.py

```diff
@@ -0,0 +1,28 @@
+# Generated by Django 4.2.1 on 2023-06-12 14:47
+
+from django.db import migrations
+
+
+def convert_min_max_to_block_version(apps, schema_editor):
+    Block = apps.get_model('blocklist', 'Block')
+    BlockVersion = apps.get_model('blocklist', 'BlockVersion')
+    File = apps.get_model('files', 'File')
+
+    for block in Block.objects.all():
```

Comment:
That call will likely fail without `.iterator()` since Django would try to cache all the rows for that queryset. And even with it... That's a lot of rows to process in a migration.

Starting to think we should split this PR into 2:
1. Introduce `BlockVersion`, maintain it as blocks are created/edited/deleted, but do not sure it as the source of truth yet, keep relying on min/max range. Migrate existing blocks using a command that we'd run after the push.
2. Replace min/max range with `BlockVersion` entirely next push (the bulk of what this PR currently contains)

What do you think ?

Reply:
> That call will likely fail without .iterator() since Django would try to cache all the rows for that queryset. 

could also make it a `values_list` and only return `id`, `guid`, `min_version`, and `max_version`.

> Starting to think we should split this PR into 2:
>
> 1.     Introduce BlockVersion, maintain it as blocks are created/edited/deleted, but do not sure it as the source of truth yet, keep relying on min/max range. Migrate existing blocks using a command that we'd run after the push.
> 2.     Replace min/max range with BlockVersion entirely next push (the bulk of what this PR currently contains)

It's possible (at some currently unquantifiable amount of effort) - I doubt the second pr will be significantly smaller than this pr is now though.

Reply:
> > That call will likely fail without .iterator() since Django would try to cache all the rows for that queryset.
> 
> could also make it a `values_list` and only return `id`, `guid`, `min_version`, and `max_version`.

`.iterator()` also internally does chunking to avoid fetching too much from the database driver layer at the same time, so we likely need it. We could `values_list()` on top of that yeah, but I'm not sure it's that important with `.iterator()`.

Reply:
resolved in #20837